pyquil/quilbase.py

```diff
@@ -678,7 +678,7 @@ def num_args(self) -> int:
 
     @property
     def matrix(self) -> np.ndarray:
-        to_py_matrix = np.vectorize(_convert_to_py_expression)
+        to_py_matrix = np.vectorize(_convert_to_py_expression, otypes=["O"])
```

Comment:
What is the "O" otype? can't find any docs on it

Reply:
Can't link to the subheader directly, but scroll down a bit to "Array-protocol type strings" from [here](https://numpy.org/doc/stable/reference/arrays.dtypes.html#specifying-and-constructing-data-types). `'O'` is a Python object.